extensions/adblock/subscription.vala

```diff
@@ -355,7 +355,21 @@ namespace Adblock {
                 debug ("%s for %s (%s)", directive.to_string (), request_uri, page_uri);
                 return directive;
             }
-            return null;
+
+            try {
+                //The uri is either Allowed(whitelist), Blocked(pattern), or neither
+                directive = whitelist.match (request_uri, page_uri);
+                if (directive == null) {
+                    directive = pattern.match (request_uri, page_uri);
```

Comment:
Nice work!

This could be improved slightly by going through all `features` which is a list that contains the whitelist, pattern and keys and it's the base class that has the `match` function.

Would you mind updating it to check all features in a loop instead?

Although if you don't have time ATM I'm okay merging this as-is so we get the fix in for the release in a few days and it can be improved later.

Reply:
I will check it, I cannot promise you much because I don't fully get the code. XP

Reply:
No worries. Let's get this in as-is and revisit after the release! If you're on [Telegram](http://midori-browser.org/telegram) or IRC (#midori on Freenode) we can discuss the code there